cime_config/acme/machines/template.case.run

```diff
@@ -1,416 +1,327 @@
+
+###############################################################################
+def resubmitCheck(case):
+###############################################################################
+
+    # check to see if we need to do resubmission from this particular job,
+    # Note that Mira requires special logic
+
+    caseroot     = case.get_value("CASEROOT")
+    dout_s       = case.get_value("DOUT_S")
+    mach         = case.get_value("MACH")
+    testcase     = case.get_value("TESTCASE")
+    resubmit_num = int(case.get_value("RESUBMIT"))
+    
+    # If dout_s is True than short-term archiving handles the resubmit
+    # that is not the case on Mira
+    resubmit = False
+    if not dout_s and resubmit_num > 0: 
+        resubmit = True
+    elif dout_s and mach == 'mira':
+        resubmit = True
+
+    if resubmit:
+	cmd = os.path.join(caseroot,"case.submit") + "-resubmit -scriptname case."
+        if testcase is not None and testcase == 'ERR':
+	   cmd = cmd + "test"
+        else:
+	   cmd = cmd + "run"
+        logger.debug( "running resubmit check %s " %cmd)
+        rc, output, error = run_cmd(cmd, ok_to_fail=True, from_dir=caseroot)
+        if rc != 0:
+            logger.warn("could not run %s submit script, rc=%s, output=%s, error=%s" %(model,rc,output,error))
+
+###############################################################################
+def DoDataAssimilation(case, da_script, lid):
+###############################################################################
+
+    cmd = "da_script 1> da.log.%s 2>&1" %(lid)
+    run_cmd(cmd)
+    cmd = "sleep(1)"
+    run_cmd(cmd)
+    disposeLog(case, 'da', lid)
+
+###############################################################################
+def appendCaseStatus(caseroot, msg):
+###############################################################################
+
+    with open(os.path.join(caseroot,"CaseStatus"), "a") as fd:
+        fd.write(msg)
+
+###############################################################################
+def _main_func():
+###############################################################################
+
+    # Set up the run, run the model, do the postrun steps
+
+    parse_command_line(sys.argv)
+
+    case = Case()
+
+    run_with_submit = case.get_value("RUN_WITH_SUBMIT")
+    expect (run_with_submit,
+             "You are not calling the run script via the submit script. "
+             "As a result, short-term archiving will not be called automatically."
+             "Please submit your run by runnint the submit script like so:"
+            " ./case.submit")
+
+    data_assimilation        = case.get_value("DATA_ASSIMILATION")
+    data_assimilation_cycles = case.get_value("DATA_ASSIMILATION_CYCLES")
+    data_assimilation_script = case.get_value("DATA_ASSIMILATION_SCRIPT")
+
+    # set up the LID 
+    lid = time.strftime("%y%m%d-%H%M%S")
+    os.environ["LID"] = lid
+
+    for cycle in range(0,data_assimilation_cycles):
```

Comment:
The zero as first argument to range is not necessary. Python assumes zero is the start if you only provide one argument.